### PR TITLE
Reword “Extend kubectl with plugins”

### PR DESCRIPTION
**Pull request by sftim**
Reword https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/

- Drop text about kubectl plugin beta status etc
- Tweak page to bring it closer to website style guide recommendations

Deploy [preview](https://deploy-preview-17005--kubernetes-io-master-staging.netlify.com/docs/tasks/extend-kubectl/kubectl-plugins/)

### CODE REVIEW

**Comment by zparnold** [comment]
@sftim can you please resolve the merge conflicts?
Otherwise I'm happy to get this merged.

File: content/en/docs/tasks/extend-kubectl/kubectl-plugins.md [hunk 1]
```diff
@@ -33,7 +27,7 @@ Plugins were officially introduced as an alpha feature in the v1.8.0 release. Th
 A plugin is nothing more than a standalone executable file, whose name begins with `kubectl-`. To install a plugin, simply move this executable file to anywhere on your PATH.
 
 {{< note >}}
-Kubernetes does not provide a package manager or anything similar to install or update plugins. It is your responsibility to ensure that plugin executables have a filename that begins with `kubectl-`, and that they are placed somewhere on your PATH.
+Kubernetes does not provide a package manager or anything similar to install or update plugins. It is your responsibility to ensure that plugin executables have a filename that begins with `kubectl-`, and that they are placed somewhere on yyour PATH.
```
Comment (bradtopol) [inline content/en/docs/tasks/extend-kubectl/kubectl-plugins.md]: Found a typo yyour

**Comment by bradtopol** [comment]
assigning @zparnold since he is already reviewing
/assign @zparnold

**Comment by sftim** [comment]
I could also make the [example plugin](https://deploy-preview-17005--kubernetes-io-master-staging.netlify.com/docs/tasks/extend-kubectl/kubectl-plugins/#example-plugin) be a downloadable code example, if people like the sound of that.

File: content/en/docs/tasks/extend-kubectl/kubectl-plugins.md [hunk 2]
```diff
@@ -147,15 +143,19 @@ For example, a plugin that wishes to be invoked whenever the command `kubectl fo
...
-would be `kubectl-foo-bar-baz-arg1`. Upon not finding that plugin, it then treats the last dash-separated value as an argument (`arg1` in this case), and attempts to find the next longest possible name, `kubectl-foo-bar-baz`.
-Upon finding a plugin with this name, it then invokes that plugin, passing all args and flags after its name to the plugin executable.
+kubectl plugins must parse and validate all of the arguments passed to them.
+See [using the command line runtime package](#using-the-command-line-runtime-package) for details of a Go library aimed at plugin authors.
+
+Here are some additional cases where users invoke your plugin while providing additional flags and arguments. This builds upon the the `kubectl-foo-bar-baz` plugin from the scenario above.
+
+If you run `kubectl foo bar baz arg1 --flag=value arg2`, kubectl's plugin mechanism will first try to find the plugin with the longest possible name, which in this case
+would be `kubectl-foo-bar-baz-arg1`. Upon not finding that plugin, kubectl` then treats the last dash-separated value as an argument (`arg1` in this case), and attempts to find the next longest possible name, `kubectl-foo-bar-baz`.
```
Comment (tengqm) [inline content/en/docs/tasks/extend-kubectl/kubectl-plugins.md]: looks like we miss a back tick on line 157.

File: content/en/docs/tasks/extend-kubectl/kubectl-plugins.md [hunk 3]
```diff
@@ -164,10 +164,10 @@ Example:
 echo -e '#!/bin/bash\n\necho "My first command-line argument was $1"' > kubectl-foo-bar-baz
 sudo chmod +x ./kubectl-foo-bar-baz
 
-# "install" our plugin by placing it on our PATH
+# "install" your plugin by moving it to a directory in your PATH
 sudo mv ./kubectl-foo-bar-baz /usr/local/bin
 
-# ensure our plugin is recognized by kubectl
+# chck that kubectl recognizes your plugin
```
Comment (tengqm) [inline content/en/docs/tasks/extend-kubectl/kubectl-plugins.md]: chck -> check

File: content/en/docs/tasks/extend-kubectl/kubectl-plugins.md [hunk 4]
```diff
@@ -202,26 +202,26 @@ sudo chmod +x ./kubectl-foo_bar
 # move the plugin into your PATH
 sudo mv ./kubectl-foo_bar /usr/local/bin
 
-# our plugin can now be invoked from `kubectl` like so:
+# You can now be invoked your plugin via kubectl, like so:
```
Comment (tengqm) [inline content/en/docs/tasks/extend-kubectl/kubectl-plugins.md]: You can now invoke your plugin ...

File: content/en/docs/tasks/extend-kubectl/kubectl-plugins.md [hunk 5]
```diff
@@ -247,13 +247,13 @@ The following kubectl-compatible plugins are available:
...
+A way to resolve this issue is to ensure that the location of the plugin that you wish to use with `kubectl` always comes first in your `PATH`. For example, if you want to always use `/usr/local/bin/moreplugins/kubectl-foo` anytime that the `kubectl` command `kubectl foo` was invoked, change the value of your PATH to be `PATH=/usr/local/bin/moreplugins:/usr/local/bin/plugins`.
 
 #### Invocation of the longest executable filename
 
-There is another kind of overshadowing that can occur with plugin filenames. Given two plugins present in a user's PATH `kubectl-foo-bar` and `kubectl-foo-bar-baz`, the `kubectl` plugin mechanism will always choose the longest possible plugin name for a given user command. Some examples below, clarify this further:
+There is another kind of overshadowing that can occur with plugin filenames. Given two plugins present in a user's `PATH` `kubectl-foo-bar` and `kubectl-foo-bar-baz`, the `kubectl` plugin mechanism will always choose the longest possible plugin name for a given user command. Some examples below, clarify this further:
```
Comment (tengqm) [inline content/en/docs/tasks/extend-kubectl/kubectl-plugins.md]: add a "," after `PATH`?

**Comment by sftim** [comment]
Good spots.
/retitle [WIP] Reword “Extend kubectl with plugins”

**Pull request merged**